scripts/Deploy.s.sol

```diff
@@ -32,133 +32,141 @@ contract Deploy is Script {
-        console.log("IntentSource :", ctx.intentSource);
-    }
+    function run() external {
+        // Initialize the deployment context struct with environment variables
+        DeploymentContext memory ctx;
+        ctx.salt = vm.envBytes32("SALT");
+        ctx.mailbox = vm.envAddress("MAILBOX");
+        ctx.deployFilePath = vm.envString("DEPLOY_FILE");
+        ctx.deployer = vm.rememberKey(vm.envUint("PRIVATE_KEY"));
+
+        // Compute salts for each contract
+        ctx.intentSourceSalt = getContractSalt(ctx.salt, "INTENT_SOURCE");
+        ctx.inboxSalt = getContractSalt(ctx.salt, "INBOX");
+        ctx.hyperProverSalt = getContractSalt(ctx.salt, "HYPER_PROVER");
+
+        vm.startBroadcast();
+
+        // Deploy deployer if it hasn't been deployed
+        if (!isDeployed(address(create3Deployer))) {
+            deployCreate3Deployer();
+            console.log(
+                "Deployed Create3Deployer : ",
+                address(create3Deployer)
+            );
+        }
 
-    // Deploy Inbox - using a code block to manage variable lifetimes
-    {
-        // Initialize solvers array properly with a defined length
-        address[] memory solvers = new address[](0);
-        uint96 minBatcherReward = 0;
-        ctx.inboxConstructorArgs = abi.encode(ctx.deployer, true, minBatcherReward, solvers);
-        
-        bytes memory inboxBytecode = abi.encodePacked(
-            type(Inbox).creationCode,
-            ctx.inboxConstructorArgs
-        );
-        
-        bool wasInboxDeployed;
-        (ctx.inbox, wasInboxDeployed) = deployWithCreate3(
-            inboxBytecode,
-            ctx.deployer,
-            ctx.inboxSalt
-        );
-        
-        console.log("Inbox :", ctx.inbox);
+        // Deploy IntentSource - using a code block to manage variable lifetimes
+        {
+            bool deployed;
+            (ctx.intentSource, deployed) = deployWithCreate3(
+                type(IntentSource).creationCode,
+                ctx.deployer,
+                ctx.intentSourceSalt
+            );
+            console.log("IntentSource :", ctx.intentSource);
+        }
+
+        // Deploy Inbox - using a code block to manage variable lifetimes
+        {
+            // Initialize solvers array properly with a defined length
+            address[] memory solvers = new address[](0);
+            uint96 minBatcherReward = 0;
+            ctx.inboxConstructorArgs = abi.encode(
+                ctx.deployer,
+                true,
+                minBatcherReward,
+                solvers
+            );
+
+            bytes memory inboxBytecode = abi.encodePacked(
+                type(Inbox).creationCode,
+                ctx.inboxConstructorArgs
+            );
+
+            bool wasInboxDeployed;
+            (ctx.inbox, wasInboxDeployed) = deployWithCreate3(
+                inboxBytecode,
+                ctx.deployer,
+                ctx.inboxSalt
+            );
 
-        if (!wasInboxDeployed) {
```

Comment:
The deployment script has changed but doesn't provide a clear upgrade path for existing deployed contracts. Consider adding migration utilities or scripts.